CONTRIBUTING.md

```diff
@@ -0,0 +1,123 @@
+
+#### Hot reloading
+
+When running `npm run dev`, you start a Next.js dev server on `http://localhost:3000`, which includes hot reloading out of the box. Make changes on any of the files in `src` and see the changes immediately.
+
+> NOTE: When working on CSS, you will need to manually refresh the page after changes. (Improving this through a PR is very welcome!)
+
+#### Databases
+
+Included is a Docker Compose file that starts up MySQL, Postgres, and MongoDB databases on localhost.
+
+It will use port `3306`, `5432`, and `27017` on localhost respectively; please make sure those ports are not used by other services on localhost.
+
+You can start them with `npm run db:start` and stop them with `npm run db:stop`.
+
+You will need Docker and Docker Compose installed to be able to start / stop the databases.
+
+When stopping the databases, it will reset their contents.
+
+#### Testing
+
+Tests can be run with `npm run test`.
+
+Automated tests are currently crude and limited in functionality, but improvements are in development.
+
+Currently, to run tests you need to first have started local test databases (e.g. using `npm run db:start`).
+
+The databases can take a few seconds to start up, so you might need to give it a minute before running the tests.
+
+## For maintainers
+
+We use [semantic-release](https://github.com/semantic-release/semantic-release) together with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0) to automate releases. This makes the maintainenance process easier and less error-prone to human error. Please study the "Conventional Commits" site to understand how to write a good commit message.
+
+When accepting Pull Requests, make sure the following:
+
+- Use "Squash and merge"
+- Make sure you merge contributor PRs into `canary`
+- Rewrite the commit message to conform to the `Conventional Commits` style. Check the "Recommended Scopes" section for further advice.
+- Optionally link issues the PR will resolve (You can add "close" in front of the issue numbers to close the issues automatically, when the PR is merged. `semantic-release` will also comment back to connected issues and PRs, notifying the users that a feature is added/bug fixed, etc.)
+
+### Recommended Scopes
+
+A typical conventional commit looks like this:
+
+```
+type(scope): title
+
+body
+```
+
+Scope is the part that will help groupping the different commit types in the release notes.
+
+Some recommened scopes are:
+
+- **provider** - Provider related changes. (eg.: "feat(provider): add X provider", "docs(provider): fix typo in X documentation"
+- **adapter** - Adapter related changes. (eg.: "feat(adapter): add X provider", "docs(provider): fix typo in X documentation"
+- **db** - Database related changes. (eg.: "feat(db): add X database", "docs(db): fix typo in X documentation"
+- **deps** - Adding/removing/updating a dependency (eg.: "chore(deps): add X")
+
```

Comment:
- We don't use `semantic-release`
- all the scope would just be **adapter**, so we may need more specific ones. I am not sure how the release notes will look for a `lerna` release. Will it group things by package name by default and only include packages that are changed etc?
- not mentioned in the main repo, but merging from `canary` to `main` should probably not be a "Squash and merge", otherwise the release note will only include a simple line (of one commit). It is good to squash and merge on users' commits to cut the noise/keep it simple, but the changes to `main` are (hopefully) already nicely formatted commits anyway, focusing on a single change/feature/bugfix

Reply:
> Will it group things by package name by default and only include packages that are changed etc?
Yep, it should do. This is why I want to start testing out the release pipeline